docker/docker_client.go

```diff
@@ -332,14 +334,9 @@ func (c *dockerClient) ping() (*pingResponse, error) {
 		}
 		return pr, nil
 	}
-	scheme := "https"
-	pr, err := ping(scheme)
-	if err != nil {
-		scheme = "http"
-		pr, err = ping(scheme)
-		if err == nil {
-			return pr, nil
-		}
+	pr, err := ping("https")
+	if err != nil && c.ctx.DockerInsecureSkipTLSVerify {
```

Comment:
This LGTM, and I believe docker upstream is not doing this since the code above was from Docker itself (I'll have to confirm this, I'm afk right now, please do check upstream if it's something worth)